kernel-spark/src/main/java/io/delta/kernel/spark/snapshot/unitycatalog/UCManagedTableSnapshotManager.java

```diff
@@ -57,35 +61,90 @@ public UCManagedTableSnapshotManager(
-    throw new UnsupportedOperationException(
-        "UCManagedTableSnapshotManager.loadLatestSnapshot is not yet implemented");
+    return ucCatalogManagedClient.loadSnapshot(
+        engine,
+        tableId,
+        tablePath,
+        Optional.empty() /* versionOpt */,
+        Optional.empty() /* timestampOpt */);
   }
 
   @Override
   public Snapshot loadSnapshotAt(long version) {
-    throw new UnsupportedOperationException(
-        "UCManagedTableSnapshotManager.loadSnapshotAt is not yet implemented");
+    return ucCatalogManagedClient.loadSnapshot(
+        engine, tableId, tablePath, Optional.of(version), Optional.empty() /* timestampOpt */);
   }
 
+  /**
+   * Finds the active commit at a specific timestamp.
+   *
+   * <p>For UC-managed tables, this loads the latest snapshot and uses {@link
+   * DeltaHistoryManager#getActiveCommitAtTimestamp} to resolve the timestamp to a commit.
+   */
   @Override
   public DeltaHistoryManager.Commit getActiveCommitAtTime(
       long timestampMillis,
       boolean canReturnLastCommit,
       boolean mustBeRecreatable,
       boolean canReturnEarliestCommit) {
-    throw new UnsupportedOperationException(
-        "UCManagedTableSnapshotManager.getActiveCommitAtTime is not yet implemented");
+    SnapshotImpl snapshot = (SnapshotImpl) loadLatestSnapshot();
+    List<ParsedCatalogCommitData> catalogCommits = snapshot.getLogSegment().getAllCatalogCommits();
+    return DeltaHistoryManager.getActiveCommitAtTimestamp(
+        engine,
+        snapshot,
+        snapshot.getLogPath(),
+        timestampMillis,
+        mustBeRecreatable,
+        canReturnLastCommit,
+        canReturnEarliestCommit,
+        catalogCommits);
   }
 
+  /**
+   * Checks if a specific version exists and is accessible.
+   *
+   * <p>For UC-managed tables with CCv2, log files may be cleaned up, so we need to use
+   * DeltaHistoryManager to find the earliest available version based on filesystem state.
+   */
   @Override
-  public void checkVersionExists(long version, boolean mustBeRecreatable, boolean allowOutOfRange) {
-    throw new UnsupportedOperationException(
-        "UCManagedTableSnapshotManager.checkVersionExists is not yet implemented");
+  public void checkVersionExists(long version, boolean mustBeRecreatable, boolean allowOutOfRange)
+      throws VersionNotFoundException {
+    // Load latest to get the current version bounds
+    SnapshotImpl snapshot = (SnapshotImpl) loadLatestSnapshot();
+    long latestRatifiedVersion = snapshot.getVersion();
+
+    // Fast path: check upper bound before expensive filesystem operations
+    if ((version > latestRatifiedVersion) && !allowOutOfRange) {
+      throw new VersionNotFoundException(version, 0, latestRatifiedVersion);
+    }
+
+    // Compute earliestRatifiedCommitVersion from catalog commits
+    List<ParsedCatalogCommitData> catalogCommits = snapshot.getLogSegment().getAllCatalogCommits();
+    Optional<Long> earliestRatifiedCommitVersion =
```

Comment:
It's not clear to me how this is helpful or what this variable is for.

(1) Every commit in the delta log was at one point ratified, so `ealiestRatifiedCommitVersion` is a bit confusing.

(2) I think you just mean: Earliest catalog commit. But, a catalog commit can also be published. So the delta log can have published deltas 0 to 20, and the catalog can have commits 10 to 30, and here the earliestCatalogCommit will be 10 -- so what? why is that helpful?

Reply:
Dah. I see `earliestRatifiedCommitVersion` is a Kernel term. Ha. I'll ask @allisonport-db to PTAL here

Reply:
I see .. it's just used for a version 0 edge case